### PR TITLE
General eventing and kafka updates

### LINKED ISSUE #3078: Apache Kafka Channel Example(docs/eventing/samples/kafka/channel/index.md) simplification
The rendered content on the website isn't as friendly to navigate as the repo

## Expected Behavior
The files are rendered inline like the previous resources that are created or there is a hyperlink to the files to make the experience for users on the site more friendly

## Actual Behavior
The [demo](https://knative.dev/docs/eventing/samples/kafka/channel/#creating-a-service-and-trigger-to-use-the-apache-kafka-broker) section references a bunch of files such as `000-ksvc.yaml`

There is no reference to indicate where to fetch the files. 

## Steps to Reproduce the Problem

1. Visit the [Kafka examples on Knative.dev](https://knative.dev/docs/eventing/samples/kafka/channel/#creating-a-service-and-trigger-to-use-the-apache-kafka-broker)
1. A user will not be able to complete the examples without knowing about [the docs repo](https://github.com/knative/docs/tree/master/docs/eventing/samples/kafka/channel) to find the referenced files

**Install information:**
 - Platform (GKE, IKS, AKS, etc.): n/a
 - Knative Version: 0.19 / master

### LINKED ISSUE #3783: There is a mistake in the installation procedure of Strimzi.
There is a mistake in the installation procedure of Strimzi.

The API version of Strimizi has been changed in the following commit.
https://github.com/knative/docs/pull/3603/files

However, the version of Strimzi Operator used in the Strimzi installation procedure has not been changed, so following the procedure will result in a validation error.
https://github.com/knative/docs/tree/83877ca0f686886bf036288e7712bd884303de8e/docs/eventing/samples/kafka#setting-up-apache-kafka

Specifically, the following error will occur.
`no matches for kind "Kafka" in version "kafka.strimzi.io/v1beta2`

In addition, the configuration file itself has not been able to keep up with the incompatible changes, and parts that were changed to be written in `array` are still in `map`.

See the `Redesign of the .spec.kafka.listeners section` in the following document.
https://github.com/strimzi/strimzi-kafka-operator/releases/tag/0.20.0

## Expected Behavior

can be installed

## Actual Behavior

cannot be installed

### PR DESCRIPTION
**Pull request by abrennan89**
Fixes #3783
Fixes #3078

## Proposed Changes

- Remove outdated Strimzi install and point to Strimzi quickstart docs instead
- Remove yaml sample files in samples/kafka, and references to same, and replace them with including the yaml directly in the doc instead
- Remove no-copy from verification command in Eventing installation docs
- Replace link to Kafka channel sample with link to knative-sandbox Kafka channel. Not sure on this last one, confirm with eng what's our recommended way to install a Kafka channel. We can maybe remove some of the channel sample docs altogether in this case or replace the YAML samples with what's in sandbox ⚠️  cc @devguyio I'd love your input on this.

### CODE REVIEW

File: docs/eventing/samples/kafka/channel/README.md
Comment (abrennan89) [inline docs/eventing/samples/kafka/channel/README.md:80]: @devguyio this doesn't really look Kafka related at all. I feel like we should remove it and write a proper procedure for what is trying to be achieved by this one instead? I'm confused by the use of event source etc here.

**Comment by abrennan89** [comment]
If this PR looks like an improvement and doesn't break anything / make anything incorrect, I'd like to get it merged and not hold it up too much on trying to make it perfect. I'd like to move these samples into the main dev/admin guides and just wanted to try to clean it up a bit first.

**Comment by abrennan89** [comment]
Added a bunch of reviewers but for now I think this just needs and SME review and possibly a lgtm when it's ready.

File: docs/eventing/samples/kafka/channel/README.md
Comment (xtreme-sameer-vohra) [inline docs/eventing/samples/kafka/channel/README.md:82]: There is some missing config as the broker was using `InMmemoryChannel` and it had an annotation of `MTChannelBasedBroker`

See `Set as default broker implementation` > https://knative.dev/v0.23-docs/eventing/broker/kafka-broker/

**Review by xtreme-sameer-vohra — CHANGES_REQUESTED** [review]
Thanks Ashleigh. Added few comments
Disclaimer: I didn't check the auth sections (TLS, etc)

**Comment by xtreme-sameer-vohra** [comment]
Hehe not sure why prow assigned me 😁

File: docs/eventing/samples/kafka/channel/README.md
Comment (snneji) [inline docs/eventing/samples/kafka/channel/README.md:267]: A lot of parentheses in this section. Not sure if they are all necessary.

File: docs/eventing/samples/kafka/channel/README.md
Comment (snneji) [inline docs/eventing/samples/kafka/channel/README.md:298]:
```suggestion
        The default secret name and namespace are `kafka-cluster` and `knative-eventing` respectively. If you reference a secret in a different namespace, make sure you configure your roles and bindings so that the `knative-eventing` Pods can access it.
```

Comment (snneji) [reply docs/eventing/samples/kafka/channel/README.md:298]: Kubernetes docs capitalise Pods

File: docs/eventing/samples/kafka/channel/README.md
Comment (snneji) [inline docs/eventing/samples/kafka/channel/README.md:183]: Should there be `kubectl apply` commands here, as stated in [Documenting YAML](https://knative.dev/docs/help/contributor/style-guide/documenting-code/#documenting-yaml) in the Style Guide?

**Review by snneji — CHANGES_REQUESTED** [review]

File: docs/eventing/samples/kafka/channel/README.md
Comment (abrennan89) [reply docs/eventing/samples/kafka/channel/README.md:298]: I don't think we have been until this point but that's OK, can you maybe add this to the word list or create an issue to add it if it's not there already?

File: docs/eventing/samples/kafka/channel/README.md
Comment (abrennan89) [reply docs/eventing/samples/kafka/channel/README.md:267]: Probably not, I'd say none of them are, but per my comment above https://github.com/knative/docs/pull/4028#issuecomment-881791577 I'm not trying to edit this to perfection in this PR - the primary reason for it is to fix the three issues linked, so for things that need additional rewriting like this or the kubectl commands as you mention in the other comment below, I'd prefer if you could raise another issue for us to clean up that stuff later so we're not blocking this PR on it. That OK?

File: docs/eventing/samples/kafka/channel/README.md
Comment (abrennan89) [reply docs/eventing/samples/kafka/channel/README.md:183]: see https://github.com/knative/docs/pull/4028#discussion_r675155520

File: docs/eventing/samples/kafka/channel/README.md
Comment (abrennan89) [inline docs/eventing/samples/kafka/channel/README.md:20]:
```suggestion
    spec:
```

File: docs/eventing/samples/kafka/channel/README.md
Comment (abrennan89) [inline docs/eventing/samples/kafka/channel/README.md:22]:
```suggestion
      numPartitions: 3
      replicationFactor: 1
```

File: docs/eventing/samples/kafka/channel/README.md
Comment (abrennan89) [inline docs/eventing/samples/kafka/channel/README.md:42]:
```suggestion
        spec:
```

File: docs/eventing/samples/kafka/channel/README.md
Comment (abrennan89) [inline docs/eventing/samples/kafka/channel/README.md:44]:
```suggestion
          numPartitions: 3
          replicationFactor: 1
```

File: docs/eventing/samples/kafka/channel/README.md
Comment (abrennan89) [reply docs/eventing/samples/kafka/channel/README.md:82]: See above comment - idk if we should even keep this and try to fix it or remove for now

File: docs/eventing/samples/kafka/channel/README.md
Comment (xtreme-sameer-vohra) [reply docs/eventing/samples/kafka/channel/README.md:80]: I'll defer to @devguyio as I don't have a lot of context on use cases. From a strictly completeness point of view, I can see the value in an example that shows `Configuring the Knative Broker for use with Apache Kafka Channels`, but the example is not working in its current state and requires some TLC.

File: docs/eventing/samples/kafka/channel/README.md
Comment (devguyio) [reply docs/eventing/samples/kafka/channel/README.md]: I think there's a small confusion here and this was initially correct, `spec` here is the spec of the cluster default so it is should be at the same level as apiVersion and Kind, unlike the above one which is the actual `KafkaChannel` . Check https://knative.dev/docs/eventing/channels/channel-types-defaults/#channel-types-and-defaults . 

So the correct yaml is
```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: default-ch-webhook
  namespace: knative-eventing
data:
  # Configuration for defaulting channels that do not specify CRD implementations.
  default-ch-config: |
    clusterDefault:
      apiVersion: messaging.knative.dev/v1beta1
      kind: KafkaChannel
      spec:
        numPartitions: 3
        replicationFactor: 1
```

**Review by xinydev — COMMENTED** [review]
It seems that these two files also need to be updated, or should we simply delete them?

docs/eventing/samples/kafka/README.md  docs/eventing/samples/kafka/kafka.yaml

File: docs/eventing/samples/kafka/channel/README.md
Comment (Shashankft9) [inline docs/eventing/samples/kafka/channel/README.md:9]: can the link be replaced with https://knative.dev/docs/admin/install/eventing/install-eventing-with-yaml/#optional-install-a-default-channel-messaging-layer instead so that it directly lands on the page where a channel layer is installed?

and also one more prerequisite of installing the broker layer since its being used in this sample? (mt channel based) https://knative.dev/docs/admin/install/eventing/install-eventing-with-yaml/#optional-install-a-broker-layer

File: docs/eventing/samples/kafka/channel/README.md
Comment (Shashankft9) [reply docs/eventing/samples/kafka/channel/README.md:82]: When I was trying this sample a week ago, I actually liked that all of these pieces are in one place, so that I can just follow the steps and see the complete picture without switching in different tabs and getting confused. Even though its not kafka related as you mentioned in your comment above, I think its better to justify with examples here that kafka is actually being used to do _these_ things - just my opinion. 

If the broker example stays, I think this is the configmap change that has to go in as @xtreme-sameer-vohra mentioned, because otherwise it had InMemoryChannel in there which would essentially create imc for all the brokers. But now since we have already changed the default Channel to be a Kafka channel in earlier steps, we can just simply put `Kind: Channel`
```
kubectl apply -f - <<EOF
---
apiVersion: v1
kind: ConfigMap
metadata:
  name: config-br-default-channel
  namespace: knative-eventing
data:
  channelTemplateSpec: |
    apiVersion: messaging.knative.dev/v1
    kind: Channel
EOF
```

And related to this, I think you removed both the steps - `This will give you two pods, such as:` and `Inside the Apache Kafka cluster you should see two new topics, such as:`. Here I think deleting the pods part makes sense, but about topic creation, it should stay but it should be changed to:
```
Inside the Apache Kafka cluster you should see one new topic, such as:
knative-messaging-kafka.default.default-kne-trigger
```

File: docs/eventing/samples/kafka/channel/README.md
Comment (abrennan89) [reply docs/eventing/samples/kafka/channel/README.md:9]: I don't want to get too deep into adding specific links, because I have a feeling we'll move these samples to a better place and also maybe do some work on splitting out the optional stuff from the eventing install later. For now I've updated the prereqs to say you need to install eventing and the optional broker and kafka channel components. I think it's better to keep it high level for now.

File: docs/eventing/samples/kafka/channel/README.md
Comment (abrennan89) [reply docs/eventing/samples/kafka/channel/README.md:218]: Ended up just adding them since this PR has been blocked on reviews for nearly two weeks anyway 🤦🏻 🤷🏻‍♀️

File: docs/eventing/samples/kafka/channel/README.md
Comment (abrennan89) [reply docs/eventing/samples/kafka/channel/README.md]: I don't want to keep blocking the PR on this since it's just for general clean up, let's come back to this

File: docs/eventing/samples/kafka/channel/README.md
Comment (abrennan89) [reply docs/eventing/samples/kafka/channel/README.md]: fixed these back the way they were

File: docs/eventing/samples/kafka/channel/README.md
Comment (abrennan89) [reply docs/eventing/samples/kafka/channel/README.md]: I've decided to remove all of the broker stuff from this section, period.
It's too far out of scope to start adding even more ConfigMap details here, particularly when the topic is Kafka Channels, not configuring Brokers.
The plan is to move all of these types of topics to the admin guide anyway, e.g. https://knative.dev/docs/admin/eventing/broker-configuration/, so we'd be duplicating information here to have to clean it up again later.

This was supposed to be a quick PR to do some cleanup, not an entire rewrite of this sample, so I think we should try to get the changes merged so that we have some improvement for now, and we can follow it up with more PRs later to flesh out the full use cases and which other topics need to be included or referenced.

**Comment by abrennan89** [comment]
@xinydev good call, removing these since they're basically outdated Strimzi install instructions

**Comment by abrennan89** [comment]
**FYI for reviewers**: let's err on the side of removing things that are outdated and doing basic cleanup for now, and then following this PR with other PRs to add more changes. There is a lot of reorg and rewriting that needs to happen with the Kafka content in general - let's not block one PR in the hopes that **everything** is included at once, please, because it just won't happen.

**Comment by abrennan89** [comment]
@snneji I think you need to add consent for the cla bot, alternatively @thisisnotapril can you override it please? 🙏🏻

**Pull request merged**
